rest-api/tools/export_tables.py

```diff
@@ -0,0 +1,76 @@
+"""Uses Cloud SQL export API to export the contents of one more tables to CSV files.
+"""
+
+import logging
+import sys
+
+from time import sleep
+from main_util import get_parser, configure_logging
+from googleapiclient import discovery
+from oauth2client.service_account import ServiceAccountCredentials
+
+_SCOPE = "https://www.googleapis.com/auth/cloud-platform"
+_INSTANCE = "rdrmaindb"
+
+def main(args):
+  credentials = ServiceAccountCredentials.from_json_keyfile_name(args.creds_file, [_SCOPE])
+  service = discovery.build('sqladmin', 'v1beta4', credentials=credentials)
+  table_names = args.tables.split(',')
+  for table in table_names:
+    uri = args.output_path + '/%s.csv' % table
+    logging.info("Exporting %s.%s to %s..." % (args.database, table, uri))
+    request_body = {
+      "exportContext": {
+        "kind": "sql#exportContext",
+        "fileType": "CSV",
+        "uri": uri,
+        "databases": [
+          args.database
+        ],
+        "csvExportOptions": {
+          "selectQuery": "select * from %s" % table
+        }
+      }
+    }
+    request = service.instances().export(project=args.project, instance=_INSTANCE,
+                                       body=request_body)
+    response = request.execute()
+    status = response['status']
+    if status != 'DONE':
+      logging.info("Waiting for export of %s to complete..." % table)
+      while status != 'DONE':
+        sleep(1)
+        request = service.operations().get(project=args.project, operation=response['name'])
+        response = request.execute()
+        status = response['status']
+    if response.get('error'):
+      logging.error("Errors in response: " % response)
+      sys.exit(-1)
+    else:
+      logging.info('Done exporting %s.' % table)
+
+if __name__ == '__main__':
+  configure_logging()
+  parser = get_parser()
+  parser.add_argument('--project',
+                      type=str,
+                      help='Project to export data for',
+                      required=True)
+  parser.add_argument('--creds_file',
+                      type=str,
+                      help='Path to credentials JSON file.',
+                      required=True)
+  parser.add_argument('--output_path',
+                      type=str,
+                      help='GCS path to write the output files to.',
+                      required=True)
+  parser.add_argument('--database',
+                      type=str,
+                      help='Name of the database containing the tables to export.',
+                      required=True)
+  parser.add_argument('--tables',
```

Comment:
You can use action='append' to make a list directly, for example with `--table one --table two`. https://docs.python.org/3/library/argparse.html#action

Reply:
OK. Given that I have a shell script wrapper for this thing right now, it would get complicated to do this properly. I think I'm gonna keep it comma-separated if that's cool.